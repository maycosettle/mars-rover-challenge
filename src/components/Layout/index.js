export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
