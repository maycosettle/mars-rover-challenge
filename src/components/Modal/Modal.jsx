import PropTypes from 'prop-types';
import { Fragment } from 'react';
import { Dialog, Transition } from '@headlessui/react';

import { Footer } from './Footer';

import useStyles from './styles';

const Modal = ({ openModal, onClose, onSubmit, children }) => {
  const styles = useStyles();

  return (
    <Transition.Root show={openModal} as={Fragment}>
      <Dialog as="div" className={styles.mainContainer} onClose={onClose}>
        <div className={styles.container}>
          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <Dialog.Overlay className={styles.overlay} />
          </Transition.Child>

          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
            enterTo="opacity-100 translate-y-0 sm:scale-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100 translate-y-0 sm:scale-100"
            leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
          >
            {children && (
              <div className={styles.children}>
                {children}
                <Footer onClose={onClose} onSubmit={onSubmit} />
              </div>
            )}
          </Transition.Child>
        </div>
      </Dialog>
    </Transition.Root>
  );
};

Modal.propTypes = {
  openModal: PropTypes.func,
  onClose: PropTypes.func,
  onSubmit: PropTypes.func,
  children: PropTypes.node,
};

Modal.propTypes = {
  openModal: () => {},
  onClose: () => {},
  onSubmit: () => {},
};

export default Modal;
