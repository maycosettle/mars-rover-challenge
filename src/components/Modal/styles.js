const useStyles = () => ({
  mainContainer: 'fixed z-10 inset-0 overflow-y-auto',
  container: 'flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0',
  overlay: 'fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity',
  children:
    'inline-block align-bottom bg-white rounded-lg px-4 pt-5 pb-4 text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full sm:p-6',
});

export default useStyles;
