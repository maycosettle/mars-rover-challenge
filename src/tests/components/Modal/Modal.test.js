import { shallow } from 'enzyme';
import { Modal } from 'components/Modal';

describe('tests on <Modal />', () => {
  const wrapper = shallow(<Modal />);

  test('Should render Modal correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
