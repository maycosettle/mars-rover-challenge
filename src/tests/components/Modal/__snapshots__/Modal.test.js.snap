// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Modal /> Should render Modal correctly 1`] = `
<Transition
  as={Symbol(react.fragment)}
>
  <Dialog
    as="div"
    className="fixed z-10 inset-0 overflow-y-auto"
  >
    <div
      className="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0"
    >
      <Child
        as={Symbol(react.fragment)}
        enter="ease-out duration-300"
        enterFrom="opacity-0"
        enterTo="opacity-100"
        leave="ease-in duration-200"
        leaveFrom="opacity-100"
        leaveTo="opacity-0"
      >
        <Overlay
          className="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity"
        />
      </Child>
      <Child
        as={Symbol(react.fragment)}
        enter="ease-out duration-300"
        enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
        enterTo="opacity-100 translate-y-0 sm:scale-100"
        leave="ease-in duration-200"
        leaveFrom="opacity-100 translate-y-0 sm:scale-100"
        leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
      />
    </div>
  </Dialog>
</Transition>
`;
